### ISSUE TITLE
Can PascalCase components be coloured differently to lowercase?

### ISSUE THREAD

**Issue by jameshulse** [vuejs/vetur#795]
- [X] I have searched through existing issues
- [X] I have read through [docs](https://vuejs.github.io/vetur)
- [X] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

Just wondering if we can get a similar experience to what VSCode provides in JSX. Lowercase html elements are coloured differently to camelcase components.

JSX Example:
![screen shot 2018-05-17 at 13 58 29](https://user-images.githubusercontent.com/797280/40178797-6fe070da-59da-11e8-999b-03aefb1d10a1.png)

Current Vue Template:
![screen shot 2018-05-17 at 13 59 48](https://user-images.githubusercontent.com/797280/40178842-970076ba-59da-11e8-9854-d77c8adab1a7.png)

**Comment by octref** [comment]
Love the idea -- although in HTML there is no distinction between upper/lower case, this could be useful to have for Vue's template.

But technically yes, this is doable.

![image](https://user-images.githubusercontent.com/4033249/40247670-ffe9d444-5a81-11e8-8bfa-d22c2f76dbb4.png)

**Comment by jameshulse** [comment]
Hi @octref. This is still something I'd really like to see in Vetur.

If I look at implementing this myself, am I correct in thinking I might just need to duplicate the current `meta.tag.any.html` definition but with a case-sensitive regex and a new name?

**Comment by jameshulse** [comment]
I have solved this using the existing scope `entity.name.tag.other.html`. This will colour lower case and pascal case components which actually seems to be the more correct solution.

**Comment by jameshulse** [comment]
For those who want to copy paste a solution, here is what I have in my User Settings:
```json
"editor.tokenColorCustomizations": {
        "[Default Dark+]": {
            "textMateRules": [
                {
                    "scope": "entity.name.tag.other.html",
                    "settings": {
                        "foreground": "#4EC9B0"
                    }
                }
            ]
        },
    },
```
You will need to replace `Default Dark+` with the name of your theme, and the colour value obviously.

**Comment by octref** [comment]
@jameshulse What you are doing is to color non HTML tags separately. Is that what you wanted?

Looking at JSX, it seems the color come from `support.class`. If there is a color key that all theme set for "non-default HTML tags", I do want to tweak my TM grammar to generate that scope for all Vue specific tags. Let's keep this issue open.

**Comment by jameshulse** [comment]
Good point @octref. Colouring non-HTML tags has basically solved it for me. The only non-HTML tags in a .vue file tend to be vue components.

**Comment by jameshulse** [comment]
If it helps, I was investigating your TM grammar and I think it might be as easy as adding [A-Z] to the start of the capture for the tag name, something like:
```
- name: meta.tag.other.html
  begin: (</?)([A-Z][a-zA-Z0-9:-]+)
  beginCaptures:
    '1': {name: punctuation.definition.tag.begin.html}
    '2': {name: support.class}
  end: (/?>)
  endCaptures:
    '1': {name: punctuation.definition.tag.end.html}
  patterns:
  - include: '#tag-stuff'

- name: meta.tag.other.html
  begin: (</?)([a-z][a-zA-Z0-9:-]+)
  beginCaptures:
    '1': {name: punctuation.definition.tag.begin.html}
    '2': {name: entity.name.tag.other.html}
  end: (/?>)
  endCaptures:
    '1': {name: punctuation.definition.tag.end.html}
  patterns:
  - include: '#tag-stuff'
```
But I wasn't sure how to add this to tests etc.

**Comment by octref** [comment]
We have the test infrastructure for grammars now! (Thanks to @Patcher56 )

- Add a file to `test/grammar/fixture/`
- Run `yarn build:grammar`
- Run `yarn test:grammar`
- Verify the generated file in `test/grammar/results/` look fine to you
- Launch the `client` config. Make sure the Vue file looks fine to you.

Are you still interested in sending a PR?

**Comment by octref** [comment]
I fixed it in #1963.
